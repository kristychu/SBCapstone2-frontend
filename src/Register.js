import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { useDispatch } from 'react-redux';
import { register } from './actions';
import { Form, FormGroup, Label, Input, Button } from 'reactstrap';
import './Login.css';

function Register() {
  const dispatch = useDispatch();
  const INITIAL_STATE = { username:"", password:"", firstName:"", lastName:"", email:"", profileImg:"" };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const history = useHistory();

  const handleSubmit = evt => {
    evt.preventDefault();
    dispatch(register(formData))
    history.push('/user-routine')
  };

  const handleChange = evt => {
    const { name, value }= evt.target;
    setFormData(fData => ({
    ...fData,
    [name]: value
    }));
  };

  return (
    <Form>
      <FormGroup>
        <Label for="username">Username</Label>
        <Input type="text" name="username" id="username" value={formData.username} onChange={handleChange}/>
      </FormGroup>
      <FormGroup>
        <Label for="password">Password</Label>
        <Input type="password" name="password" id="password" value={formData.password} onChange={handleChange}/>
      </FormGroup>
      <FormGroup>
        <Label for="firstname">First name</Label>
        <Input type="text" name="firstName" id="firstname" value={formData.firstname} onChange={handleChange}/>
      </FormGroup>
      <FormGroup>
        <Label for="lastname">Last name</Label>
        <Input type="text" name="lastName" id="lastname" value={formData.lastname} onChange={handleChange}/>
      </FormGroup>
      <FormGroup>
        <Label for="email">Email</Label>
        <Input type="email" name="email" id="email" value={formData.email} onChange={handleChange}/>
      </FormGroup>
      <FormGroup>
        <Label for="profileImg">Profile Image</Label>
        <Input type="text" name="profileImg" id="profileImg" value={formData.profileImg} onChange={handleChange}/>
      </FormGroup>
      <Button color="primary" onClick={handleSubmit}>Submit</Button>
    </Form>
  );
}

export default Register;