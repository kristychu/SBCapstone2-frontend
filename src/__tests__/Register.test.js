import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers/rootReducer";
import { render } from "@testing-library/react";
import Register from "../Register";
import { MemoryRouter } from "react-router";

const store = createStore(reducer);

it("should render without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    </Provider>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Register />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
