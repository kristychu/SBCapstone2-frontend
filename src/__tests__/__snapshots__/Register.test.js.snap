// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class=""
        for="username"
      >
        Username
      </label>
      <input
        class="form-control"
        id="username"
        name="username"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="password"
      >
        Password
      </label>
      <input
        class="form-control"
        id="password"
        name="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="firstname"
      >
        First name
      </label>
      <input
        class="form-control"
        id="firstname"
        name="firstName"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="lastname"
      >
        Last name
      </label>
      <input
        class="form-control"
        id="lastname"
        name="lastName"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="email"
      >
        Email
      </label>
      <input
        class="form-control"
        id="email"
        name="email"
        type="email"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="profileImg"
      >
        Profile Image
      </label>
      <input
        class="form-control"
        id="profileImg"
        name="profileImg"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
